File: mRemoteV1/App/App.Info.vb
Imports System.Threading
Imports mRemote3G.My
Imports mRemote3G.Tools

Namespace App

    Namespace Info
        Public Class General
            Public Shared ReadOnly URLHome As String = "https://github.com/kmscode/mRemote3G"
            Public Shared ReadOnly URLForum As String = "https://github.com/kmscode/mRemote3G/issues"
            Public Shared ReadOnly URLBugs As String = "https://github.com/kmscode/mRemote3G/issues"
            Public Shared ReadOnly HomePath As String = Application.Info.DirectoryPath
            Public Shared EncryptionKey As String = "mR3m"
            Public Shared ReportingFilePath As String = ""
            Public Shared ReadOnly PuttyPath As String = Application.Info.DirectoryPath & "\PuTTYNG.exe"

            Public Shared ReadOnly Property UserAgent As String
                Get
                    Dim details As New List(Of String)
                    details.Add("compatible")
                    If Environment.OSVersion.Platform = PlatformID.Win32NT Then
                        details.Add(String.Format("Windows NT {0}.{1}", Environment.OSVersion.Version.Major,
                                                  Environment.OSVersion.Version.Minor))
                    Else
                        details.Add(Environment.OSVersion.VersionString)
                    End If
                    If EnvironmentInfo.IsWow64 Then details.Add("WOW64")
                    details.Add(Thread.CurrentThread.CurrentUICulture.Name)
                    details.Add(String.Format(".NET CLR {0}", Environment.Version.ToString()))
                    Dim detailsString As String = String.Join("; ", details.ToArray())

                    Return _
                        String.Format("Mozilla/4.0 ({0}) {1}/{2}", detailsString,
                                      System.Windows.Forms.Application.ProductName,
                                      System.Windows.Forms.Application.ProductVersion)
                End Get
            End Property

            Private Sub New()
            End Sub
        End Class

        Public Class Settings
#If Not PORTABLE Then

            Public Shared ReadOnly _
                SettingsPath As String = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) & "\" &
                                         System.Windows.Forms.Application.ProductName

#Else
            Public Shared ReadOnly SettingsPath As String = My.Application.Info.DirectoryPath
#End If
            Public Shared ReadOnly LayoutFileName As String = "pnlLayout.xml"
            Public Shared ReadOnly ExtAppsFilesName As String = "extApps.xml"
            Public Const ThemesFileName As String = "Themes.xml"

            Private Sub New()
            End Sub
        End Class

        Public Class UpdateApp
            Public Shared ReadOnly Property FileName As String
                Get
#If DEBUG Then
                    Return "update-debug.txt"
#End If
                    Select Case MySettingsProperty.Settings.UpdateChannel.ToLowerInvariant()
                        Case "beta"
                            Return "update-beta.txt"
                        Case "debug"
                            Return "update-debug.txt"
                        Case Else
                            Return "update.txt"
                    End Select
                End Get
            End Property

            Private Sub New()
            End Sub
        End Class

        Public Class Connections
            Public Shared ReadOnly DefaultConnectionsPath As String = Settings.SettingsPath
            Public Shared ReadOnly DefaultConnectionsFile As String = "confCons.xml"
            Public Shared ReadOnly DefaultConnectionsFileNew As String = "confConsNew.xml"
            Public Shared ReadOnly ConnectionFileVersion As Double = 2.5

            Private Sub New()
            End Sub
        End Class

        Public Class Credentials
            Public Shared ReadOnly CredentialsPath As String = Settings.SettingsPath
            Public Shared ReadOnly CredentialsFile As String = "confCreds.xml"
            Public Shared ReadOnly CredentialsFileNew As String = "confCredsNew.xml"
            Public Shared ReadOnly CredentialsFileVersion As Double = 1.0

            Private Sub New()
            End Sub
        End Class
    End Namespace

End Namespace

File: mRemoteV1/CS/App/App.Info.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

//using System.Environment;
using System.Threading;


namespace mRemoteNG.App.Info
{
	public class General
	{
		public static readonly string URLHome = "http://www.mremoteng.org/";
		public static readonly string URLDonate = "http://donate.mremoteng.org/";
		public static readonly string URLForum = "http://forum.mremoteng.org/";
		public static readonly string URLBugs = "http://bugs.mremoteng.org/";
		public static readonly string HomePath = (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.DirectoryPath;
		public static string EncryptionKey = "mR3m";
		public static string ReportingFilePath = "";
		public static readonly string PuttyPath = (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.DirectoryPath + "\\PuTTYNG.exe";
public static string UserAgent
		{
			get
			{
				List<string> details = new List<string>();
				details.Add("compatible");
				if (System.Environment.OSVersion.Platform == PlatformID.Win32NT)
				{
					details.Add(string.Format("Windows NT {0}.{1}", System.Environment.OSVersion.Version.Major, System.Environment.OSVersion.Version.Minor));
				}
				else
				{
					details.Add(System.Environment.OSVersion.VersionString);
				}
				if (Tools.EnvironmentInfo.IsWow64)
				{
					details.Add("WOW64");
				}
				details.Add(Thread.CurrentThread.CurrentUICulture.Name);
				details.Add(string.Format(".NET CLR {0}", Version.ToString()));
				string detailsString = string.Join("; ", details.ToArray());
						
				return string.Format("Mozilla/4.0 ({0}) {1}/{2}", detailsString, System.Windows.Forms.Application.ProductName, System.Windows.Forms.Application.ProductVersion);
			}
		}
	}
			
	public class Settings
	{
#if !PORTABLE
		public static readonly string SettingsPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.ApplicationData) + "\\" + (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.ProductName;
#else
		public static readonly string SettingsPath = (new Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase()).Info.DirectoryPath;
#endif
		public static readonly string LayoutFileName = "pnlLayout.xml";
		public static readonly string ExtAppsFilesName = "extApps.xml";
		public const string ThemesFileName = "Themes.xml";
	}
			
	public class Update
	{
public static string FileName
		{
			get
			{
#if DEBUG
				return "update-debug.txt";
#endif
//						
//						if ((string) (My.Settings.Default.UpdateChannel.ToLowerInvariant()) == "beta")
//						{
//							return "update-beta.txt";
//							}
//							else if ((string) (My.Settings.Default.UpdateChannel.ToLowerInvariant()) == "debug")
//							{
//								return "update-debug.txt";
//								}
//								else
//								{
//									return "update.txt";
//									}
						}
					}
				}
						
	public class Connections
	{
		public static readonly string DefaultConnectionsPath = App.Info.Settings.SettingsPath;
		public static readonly string DefaultConnectionsFile = "confCons.xml";
		public static readonly string DefaultConnectionsFileNew = "confConsNew.xml";
		public static readonly double ConnectionFileVersion = 2.5;
	}
						
	public class Credentials
	{
		public static readonly string CredentialsPath = App.Info.Settings.SettingsPath;
		public static readonly string CredentialsFile = "confCreds.xml";
		public static readonly string CredentialsFileNew = "confCredsNew.xml";
		public static readonly double CredentialsFileVersion = 1.0;
	}			
}
